### ISSUE TITLE
Versions, ChangeLog, NEWS and R-Forge

### ISSUE THREAD

**Issue by jarioksa** [vegandevs/vegan#53]
These have in common that they are linear. In contrast, git is a branching and merging web. How to combine these linear structures with git?

So far we have used version numbering in development branch to have some internal structuring in development versions. In R-Forge we had separate release branch where versions were numbered by CRAN releases. So the version numbers changed much more rapidly in in development versions: we had 10 CRAN releases in 2.0, and 41 development versions in the same time. 

Version numbering is linear. If we make a "feature fork", we cannot change versions there, but they must always be changed in the main branch after merging the feature. However, I suggest we drop the practice of frequent version changes in development version, and align the development version numbers with CRAN releases. I have two alternative numbering schemes:
1. First release is 2.2-0 and development (main) branch will be 2.2-1. We merge things we merge to the release branch and then release it as 2.2-2, renumber development to 2.2-3 and so on.
2. First release is 2.2-0 and development (main) branch will be 2.3-0. After publishing 2.2-0 in CRAN, we renumber release branch as 2.2-1 which would be a "patched" version. Then we release it as 2.2-2. 

In both schemes, only even numbered versions would be in CRAN. It may be possible (haven't checked) to have decimal minor numbers (2.2-0.9), but text is not allowed (2.2-0-patched). Separating development (github main branch), CRAN release, and patched CRAN release should be somehow guaranteed.

Another issue is ChangeLog and NEWS. Now we have two separate things for about the same purpose. The idea was that ChangeLog is more fine grained and intended for fellow developers, and NEWS are only updated for the release and intended for users. In addition, version control adds its own very fine grained logs. Three levels is too much. R manages with svn logs and NEWS which are updated continuously as features are added and bugs fixed. We could do the same and drop the ChangeLog (this is my suggestion). In particular, if we do not have frequent version number changes, ChangeLog has little advantage over release NEWS. Naturally, we must keep NEWS linear: they cannot be changed in feature forks but only in the main branch after the merge.

Last issue is R-Forge. I have googled for advice on maintaining repos in parallel in svn and in git. The most usual piece of advice is "don't!". The major problem is again that R-Forge is linear. So far I have managed to keep R-Forge in sync with git within a day or two. If we want to keep R-Forge, we could change policy so that we only sync the release branch to R-Forge. This would change rather infrequently and be linear minimizing the pain of syncing. The most important advantage of R-Forge is that they offer Windows binaries that can be installed directly with standard R tools. With this scheme, the stable versions are in CRAN, patched versions in R-Forge and unstable development in github. The first two can be installed without any special tools also in Windows (but no more in Mac in R-Forge), but github version needs special R packages plus C and Fortran compilers.
